### PR TITLE
Python: model stdlib `http.server` and `cgi`

### PR DESCRIPTION
**Pull request by RasmusWL**
This was needed to find results in Google Gruyere.

Fixes https://github.com/github/codeql-python-team/issues/60.

~Still in draft since I realized I didn't model any sinks :sweat_smile:~

### CODE REVIEW

File: python/ql/src/semmle/python/web/stdlib/Request.qll [hunk 1]
```diff
@@ -0,0 +1,126 @@
+/**
+ * Provides the sources and taint-flow for HTTP servers defined with standard library (stdlib).
+ * Specifically, we model `HttpRequestTaintSource`s from instances of `BaseHTTPRequestHandler`
+ * (or subclasses) and form parsing with `cgi.FieldStorage`.
```
Comment (tausbn) [inline python/ql/src/semmle/python/web/stdlib/Request.qll:4]: I keep reading "form" as a verb in this sentence ("We ... form parsing with ..."). Perhaps change `with` to `using`?

File: python/ql/src/semmle/python/web/stdlib/Request.qll [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+/** TaintKind for the field of a parsed HTTP form. */
+class CgiFieldStorageFieldKind extends TaintKind {
+    CgiFieldStorageFieldKind() { this = "CgiFieldStorageFieldKind" }
+
+    override TaintKind getTaintOfAttribute(string name) {
+        name = "filename" and result instanceof ExternalStringKind
+        or
+        name = "file" and result instanceof ExternalFileObject
+        or
+        name = "value" and result instanceof ExternalStringKind
```
Comment (tausbn) [inline python/ql/src/semmle/python/web/stdlib/Request.qll:124]: Perhaps roll this up together with `"filename"`?
(Having a single set expression per kind looks really nice!)

File: python/ql/src/semmle/python/web/stdlib/Request.qll (hunk 1 above)
Comment (RasmusWL) [reply python/ql/src/semmle/python/web/stdlib/Request.qll:4]: Done

File: python/ql/src/semmle/python/web/stdlib/Request.qll (hunk 2 above)
Comment (RasmusWL) [reply python/ql/src/semmle/python/web/stdlib/Request.qll]: Done

File: python/ql/src/semmle/python/web/stdlib/Request.qll [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /**
- * Provides the sources and taint-flow for HTTP servers defined with standard library (stdlib).
+ * Provides the sources and taint-flow for HTTP servers defined using the standard library (stdlib).
  * Specifically, we model `HttpRequestTaintSource`s from instances of `BaseHTTPRequestHandler`
  * (or subclasses) and form parsing with `cgi.FieldStorage`.
```
Comment (tausbn) [inline python/ql/src/semmle/python/web/stdlib/Request.qll:4]:
```suggestion
 * (or subclasses) and form parsing using `cgi.FieldStorage`.
```
That wasn't the "with" I was talking about, but never mind. :laughing: 
This is the change I had in mind.

Comment (RasmusWL) [reply python/ql/src/semmle/python/web/stdlib/Request.qll]: I like these suggestions... they make life easy for me, and make me fix the right things :smile:

**Pull request merged**